test-e2e/sync.js

```diff
@@ -160,6 +162,99 @@ test('start and stop sync', async function (t) {
   await disconnect()
 })
 
+test('auto-stop', async (t) => {
```

Comment:
would also test that listening for events on the sync api provides the expected sequence of payloads e.g `$sync.addListener('sync-state', ...)`

Reply:
Done in d4521f30edad827737858929681a69af24e9c05b.